lib/linter.js

```diff
@@ -278,19 +278,24 @@ function executeWithCLIEngine(CLIEngine, cwd, opts, text, callback) {
 exports.invoke = async function (cwd, args, text, mtime, callback) {
   process.chdir(cwd);
 
+  // Parse options as eslint by default, just so we can get the eslint path if
+  // necessary
+  const optsForPath = options_eslint.parse([0, 0].concat(args));
```

Comment:
Hm, don't you think you could do the parsing yourself with a simple `indexOf` check here? Performance of this function is very critical – it's the reason for the libraries existence actually – and the `--eslint-path` use case is a feature that's going to be used rarely.